flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/FamilyArgumentTypeStrategy.java

```diff
@@ -86,13 +86,16 @@ public Optional<DataType> inferArgumentType(
         if (Objects.equals(expectedNullability, Boolean.FALSE) && actualType.isNullable()) {
             return callContext.fail(
                     throwOnFailure,
-                    "Unsupported argument type. Expected nullable type of family '%s' but actual type was '%s'.",
+                    "Unsupported argument type. Expected NOT NULL type of family '%s' but actual type was '%s'.",
                     expectedFamily,
                     actualType);
         }
 
         // type is part of the family
         if (actualType.getTypeRoot().getFamilies().contains(expectedFamily)) {
+            if (Objects.equals(expectedNullability, Boolean.TRUE) && !actualType.isNullable()) {
```

Comment:
I think it would be useful for the issue text / PR text to give SQL examples of the reported issue and the expected behaviour - currently descriptions are very low level and require table planner knowledge. This would help users find and identify this issue at the SQL level